### ISSUE TITLE
[Bug-App]: Gradient logging issues when using torch.amp.GradScaler

### ISSUE THREAD

**Issue by gmongaras** [wandb/wandb#9092]
### Describe the bug

I am using wandb 0.19 on an Ubuntu machine.
It looks like wandb logs the scaled gradients when using a gradient scalar as described in this documentation:
https://pytorch.org/docs/stable/notes/amp_examples.html

I noticed this as my gradients were getting massive (the sudden decrease in variance is due to me resetting the grad scalar on accident).
![Image](https://github.com/user-attachments/assets/1e998599-9550-4d8e-9c18-dfe29eb6dd39)

Is there any way to change the behavior to log the unscaled gradients as that is what the optimizer uses to update the parameters of the model?

**Comment by gmongaras** [comment]
Thanks for helping me with this!

Since the gradients are logged in their scaled forms, it makes it difficult to debug as the optimizer updates the model using unscaled gradients. Do you think the default behavior can be changed to log the unscaled gradients since those are the ones used for update? Or perhaps a flag upon initialization?

I haven't looked deep at the code yet, but I am guessing the gradients are logged on the backward pass. Perhaps an option could be added to log them on the optimizer step instead which would resolve this issue and also log post-backward functions such as gradient clipping?

**Comment by Molem7b5** [comment]
Can I bump to get this re-opened? It would be very useful if we were able to set a flag or something and then have wandb log the unscaled gradients.
